cmd/eksctl-anywhere/cmd/upgradecluster.go

```diff
@@ -99,6 +99,7 @@ func (uc *upgradeClusterOptions) upgradeCluster(ctx context.Context) error {
 		deps.ClusterManager,
 		deps.FluxAddonClient,
 		deps.Writer,
+		types.ChangeDiff{},
```

Comment:
I don't think we need this as a constructor argument, you can just initialize it in the constructor body
also, I would prefer `types.NewChangeDiff()`